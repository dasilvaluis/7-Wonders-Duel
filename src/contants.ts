export const ELEMENT_MARGIN = 20;
export const BOARD_WIDTH = 1680;
export const MAX_CARDS = 20;
export const MAX_COINS_6 = 7;
export const MAX_COINS_3 = 9;
export const MAX_COINS_1 = 14;

// WebSocket events
export const CONNECT = 'connect';
export const GET_STATE = 'get_state';
export const SET_STATE = 'set_state';
export const GET_ELEMENTS = 'get_elements';
export const SET_ELEMENTS = 'set_elements';
export const MOVE_ELEMENT = 'move_element';
export const ADD_ELEMENTS = 'add_elements';
export const FLIP_ELEMENT = 'flip_element';
export const BRING_ELEMENT = 'bring_element';
export const SET_AGE = 'set_age';
export const GET_AGE = 'get_age';
export const YOU_START = 'you_start';
export const SET_SCORE = 'set_score';
export const SET_SCORES = 'set_scores';
export const GET_SCORES = 'get_scores';
