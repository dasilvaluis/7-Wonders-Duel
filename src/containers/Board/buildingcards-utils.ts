import {
  centerHorizontally,
  centerRow,
  getRowOf,
  getRandomElements,
  shuffleArray,
  movePositions,
  injectPositions,
  getElementSize
} from "../../utils";
import { Position, GameElement, ElementTypes, Age } from "../../types";
import { ELEMENT_MARGIN, MAX_CARDS } from "../../contants";
import { flattenDeep, createElement } from "../../utils";
import {
  buildings_i as buildingsIDb,
  buildings_ii as buildingsIIDb,
  buildings_iii as buildingsIIIDb,
  buildings_g as buildingsGDb
} from '../../data/buildings.json';

const moveRowVertically = (row: Array<Position>, rowIndex: number) =>
  row.map((position) => ({
    ...position,
    y: rowIndex * (getElementSize(ElementTypes.BUILDING_CARD).height / 3)
  }));

export const getAgeScheme = (age: Age) => {
  switch (age) {
    case "G":
    case "III":
      return [ 2, 3, 4, 2, 4, 3, 2 ];
    case "II":
      return [ 6, 5, 4, 3, 2 ];
    default:
      return [ 2, 3, 4, 5, 6 ];
  }
}

export const getAgeDeck = (age: Age) => {
  switch (age) {
    case "G":
      return buildingsGDb;
    case "III":
      return buildingsIIIDb;
    case "II":
      return buildingsIIDb;
    default:
      return buildingsIDb;
  }
};

export const fixThirdAgeCards = (cards: Array<GameElement>) => {
  const _cards = [ ...cards ];
  const buildingWidth = getElementSize(ElementTypes.BUILDING_CARD).width;

  _cards[9].x = cards[9].x - (buildingWidth + ELEMENT_MARGIN) / 2;
  _cards[10].x = cards[10].x + (buildingWidth + ELEMENT_MARGIN) / 2;

  return _cards;
}

export const getBuildingCardsPlacement = (scheme: Array<number>) => {
  const buildingWidth = getElementSize(ElementTypes.BUILDING_CARD).width;
  const rows = scheme.map((howMany, rowIndex) => {
    const row = getRowOf(howMany, buildingWidth);
    const rowMovedVertically = moveRowVertically(row, rowIndex);
    const rowCentered = centerRow(rowMovedVertically, howMany, buildingWidth);

    return rowCentered;
  });

  return centerHorizontally(flattenDeep<Position>(rows));
};

export const getShuffledCards = (age: Age): Array<GameElement> => {
  const deck = getAgeDeck(age);
  const ageCards = getRandomElements(deck, MAX_CARDS).map((card) => ({
    ...createElement(ElementTypes.BUILDING_CARD),
    imageFile: card.file,
    imageFileBackface: `building-${ age.toLowerCase() }-back.jpg`
  }));
  
  if (age === 'III' || age === 'G') {
    const thirdAgeCards = ageCards.slice(0, MAX_CARDS - 3)

    const guildCards = getRandomElements(buildingsGDb, 3).map((card) => ({
      ...createElement(ElementTypes.BUILDING_CARD),
      imageFile: card.file,
      imageFileBackface: `building-g-back.jpg`
    }));

    return shuffleArray([ ...thirdAgeCards, ...guildCards ])
  }

  return ageCards;
}

export const flipBuildingCards = (cards: Array<GameElement>, age: Age) => 
  cards.map((card, index) => {
    switch (age) {
      case 'I':
        return {
          ...card,
          faceDown: (2 <= index && index <= 4) || (9 <= index && index <= 13)
        };
      case 'II':
        return {
          ...card,
          faceDown: (6 <= index && index <= 10) || (15 <= index && index <= 17)
        };
      case 'III':
      case 'G':
        return {
          ...card,
          faceDown: (2 <= index && index <= 4) || (9 <= index && index <= 10) || (15 <= index && index <= 17)
        };
      default:
        return card;
    }
  });

export const getBuildingCards = (age: Age): Array<GameElement> => {
  const scheme = getAgeScheme(age);
  const cardsPlacement = getBuildingCardsPlacement(scheme);
  const cardsPlacementShifted = movePositions(cardsPlacement, {
    x: 0,
    y: ELEMENT_MARGIN * 2.5 + getElementSize(ElementTypes.BOARD).height
  });
  const shuffledCards = getShuffledCards(age);
  const cards: Array<GameElement> = injectPositions(shuffledCards, cardsPlacementShifted);
  const fixedCards = age === 'III' ? fixThirdAgeCards(cards) : cards;
  const flippedCards = flipBuildingCards(fixedCards, age);

  return flippedCards;
};
