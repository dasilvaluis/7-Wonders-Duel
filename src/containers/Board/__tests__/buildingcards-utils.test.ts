import { getAgeScheme, getBuildingCardsPlacement, getShuffledCards } from "../buildingcards-utils"
import { CARD_WIDTH } from "../../../contants";

describe('getBuildingCardsPlacement', () => {
  it('places first age cards', () => {
    expect(getBuildingCardsPlacement(getAgeScheme('I'), CARD_WIDTH)).toMatchSnapshot();
  });

  it('places second age cards', () => {
    expect(getBuildingCardsPlacement(getAgeScheme('II'), CARD_WIDTH)).toMatchSnapshot();
  });

  it('places third age cards', () => {
    expect(getBuildingCardsPlacement(getAgeScheme('III'), CARD_WIDTH)).toMatchSnapshot();
  });
});

describe('getShuffledCards', () => {
  it('shuffle first age cards', () => {
    expect(getShuffledCards('I')).toMatchSnapshot();
  });

  it('shuffle second age cards', () => {
    expect(getShuffledCards('II')).toMatchSnapshot();
  });

  it('shuffle third age/guild cards', () => {
    expect(getShuffledCards('III')).toMatchSnapshot();
    expect(getShuffledCards('G')).toMatchSnapshot();
  });
});
