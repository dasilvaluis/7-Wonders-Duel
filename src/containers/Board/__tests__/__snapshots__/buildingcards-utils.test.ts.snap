// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`getBuildingCardsPlacement places first age cards 1`] = `
Array [
  Object {
    "x": 705,
    "y": 0,
  },
  Object {
    "x": 850,
    "y": 0,
  },
  Object {
    "x": 632.5,
    "y": 62.5,
  },
  Object {
    "x": 777.5,
    "y": 62.5,
  },
  Object {
    "x": 922.5,
    "y": 62.5,
  },
  Object {
    "x": 560,
    "y": 125,
  },
  Object {
    "x": 705,
    "y": 125,
  },
  Object {
    "x": 850,
    "y": 125,
  },
  Object {
    "x": 995,
    "y": 125,
  },
  Object {
    "x": 487.5,
    "y": 187.5,
  },
  Object {
    "x": 632.5,
    "y": 187.5,
  },
  Object {
    "x": 777.5,
    "y": 187.5,
  },
  Object {
    "x": 922.5,
    "y": 187.5,
  },
  Object {
    "x": 1067.5,
    "y": 187.5,
  },
  Object {
    "x": 415,
    "y": 250,
  },
  Object {
    "x": 560,
    "y": 250,
  },
  Object {
    "x": 705,
    "y": 250,
  },
  Object {
    "x": 850,
    "y": 250,
  },
  Object {
    "x": 995,
    "y": 250,
  },
  Object {
    "x": 1140,
    "y": 250,
  },
]
`;

exports[`getBuildingCardsPlacement places second age cards 1`] = `
Array [
  Object {
    "x": 415,
    "y": 0,
  },
  Object {
    "x": 560,
    "y": 0,
  },
  Object {
    "x": 705,
    "y": 0,
  },
  Object {
    "x": 850,
    "y": 0,
  },
  Object {
    "x": 995,
    "y": 0,
  },
  Object {
    "x": 1140,
    "y": 0,
  },
  Object {
    "x": 487.5,
    "y": 62.5,
  },
  Object {
    "x": 632.5,
    "y": 62.5,
  },
  Object {
    "x": 777.5,
    "y": 62.5,
  },
  Object {
    "x": 922.5,
    "y": 62.5,
  },
  Object {
    "x": 1067.5,
    "y": 62.5,
  },
  Object {
    "x": 560,
    "y": 125,
  },
  Object {
    "x": 705,
    "y": 125,
  },
  Object {
    "x": 850,
    "y": 125,
  },
  Object {
    "x": 995,
    "y": 125,
  },
  Object {
    "x": 632.5,
    "y": 187.5,
  },
  Object {
    "x": 777.5,
    "y": 187.5,
  },
  Object {
    "x": 922.5,
    "y": 187.5,
  },
  Object {
    "x": 705,
    "y": 250,
  },
  Object {
    "x": 850,
    "y": 250,
  },
]
`;

exports[`getBuildingCardsPlacement places third age cards 1`] = `
Array [
  Object {
    "x": 705,
    "y": 0,
  },
  Object {
    "x": 850,
    "y": 0,
  },
  Object {
    "x": 632.5,
    "y": 62.5,
  },
  Object {
    "x": 777.5,
    "y": 62.5,
  },
  Object {
    "x": 922.5,
    "y": 62.5,
  },
  Object {
    "x": 560,
    "y": 125,
  },
  Object {
    "x": 705,
    "y": 125,
  },
  Object {
    "x": 850,
    "y": 125,
  },
  Object {
    "x": 995,
    "y": 125,
  },
  Object {
    "x": 705,
    "y": 187.5,
  },
  Object {
    "x": 850,
    "y": 187.5,
  },
  Object {
    "x": 560,
    "y": 250,
  },
  Object {
    "x": 705,
    "y": 250,
  },
  Object {
    "x": 850,
    "y": 250,
  },
  Object {
    "x": 995,
    "y": 250,
  },
  Object {
    "x": 632.5,
    "y": 312.5,
  },
  Object {
    "x": 777.5,
    "y": 312.5,
  },
  Object {
    "x": 922.5,
    "y": 312.5,
  },
  Object {
    "x": 705,
    "y": 375,
  },
  Object {
    "x": 850,
    "y": 375,
  },
]
`;

exports[`getShuffledCards shuffle first age cards 1`] = `
Array [
  Object {
    "faceDown": false,
    "id": "65e58df1-c891-4de2-b8ad-8d17be5ffc65",
    "imageFile": "",
    "imageFileBackface": "building-i-back.jpg",
    "name": "Scriptorium",
    "type": "building",
    "x": 0,
    "y": 0,
  },
  Object {
    "faceDown": false,
    "id": "c5de5d8f-84aa-44da-b517-b8f3fb6d3128",
    "imageFile": "",
    "imageFileBackface": "building-i-back.jpg",
    "name": "Torre de Guardia",
    "type": "building",
    "x": 0,
    "y": 0,
  },
  Object {
    "faceDown": false,
    "id": "fc543d7b-06bb-422f-bbb0-f38fa1e529f3",
    "imageFile": "",
    "imageFileBackface": "building-i-back.jpg",
    "name": "Botica",
    "type": "building",
    "x": 0,
    "y": 0,
  },
  Object {
    "faceDown": false,
    "id": "8dab32fb-1558-4467-a8d8-b8684c03292a",
    "imageFile": "",
    "imageFileBackface": "building-i-back.jpg",
    "name": "Baños",
    "type": "building",
    "x": 0,
    "y": 0,
  },
  Object {
    "faceDown": false,
    "id": "71bd3590-f976-45fc-a783-67268cb3aec6",
    "imageFile": "",
    "imageFileBackface": "building-i-back.jpg",
    "name": "Taller",
    "type": "building",
    "x": 0,
    "y": 0,
  },
  Object {
    "faceDown": false,
    "id": "37350a4d-e63e-45db-b4ad-1dfd8dd625a8",
    "imageFile": "",
    "imageFileBackface": "building-i-back.jpg",
    "name": "Campo Maderero",
    "type": "building",
    "x": 0,
    "y": 0,
  },
  Object {
    "faceDown": false,
    "id": "64a6833e-cefc-4359-a6df-4d5eb06f6522",
    "imageFile": "",
    "imageFileBackface": "building-i-back.jpg",
    "name": "Almacém de Arcilla",
    "type": "building",
    "x": 0,
    "y": 0,
  },
  Object {
    "faceDown": false,
    "id": "39bc70bd-e225-42d5-ac45-3685c990f27f",
    "imageFile": "",
    "imageFileBackface": "building-i-back.jpg",
    "name": "Almacém de Madera",
    "type": "building",
    "x": 0,
    "y": 0,
  },
  Object {
    "faceDown": false,
    "id": "341fe8ae-d99c-4467-879e-d7b89b19cb5f",
    "imageFile": "",
    "imageFileBackface": "building-i-back.jpg",
    "name": "Reserva de Arcilla",
    "type": "building",
    "x": 0,
    "y": 0,
  },
  Object {
    "faceDown": false,
    "id": "2d0a2d9a-cdd3-4dda-93dc-8b6d01d5b2b7",
    "imageFile": "",
    "imageFileBackface": "building-i-back.jpg",
    "name": "Cantera",
    "type": "building",
    "x": 0,
    "y": 0,
  },
  Object {
    "faceDown": false,
    "id": "9472fc54-d557-4b4e-8e67-e28b4ba67b37",
    "imageFile": "",
    "imageFileBackface": "building-i-back.jpg",
    "name": "Farmacia",
    "type": "building",
    "x": 0,
    "y": 0,
  },
  Object {
    "faceDown": false,
    "id": "e434776f-3f1b-42fa-b40a-8ce9789b7c7c",
    "imageFile": "",
    "imageFileBackface": "building-i-back.jpg",
    "name": "Teatro",
    "type": "building",
    "x": 0,
    "y": 0,
  },
  Object {
    "faceDown": false,
    "id": "3d4c20a5-94d2-4dbe-8ca6-d97c7ba4f288",
    "imageFile": "",
    "imageFileBackface": "building-i-back.jpg",
    "name": "Explotación de Arcilla",
    "type": "building",
    "x": 0,
    "y": 0,
  },
  Object {
    "faceDown": false,
    "id": "69f2873f-4c1a-4f1f-afbf-a7653fedf1d6",
    "imageFile": "",
    "imageFileBackface": "building-i-back.jpg",
    "name": "Prensa",
    "type": "building",
    "x": 0,
    "y": 0,
  },
  Object {
    "faceDown": false,
    "id": "bf86d090-3951-48c3-9bef-83613f27c1aa",
    "imageFile": "",
    "imageFileBackface": "building-i-back.jpg",
    "name": "Palenque",
    "type": "building",
    "x": 0,
    "y": 0,
  },
  Object {
    "faceDown": false,
    "id": "b31d61d0-bb12-45f4-a415-48f2b044c194",
    "imageFile": "",
    "imageFileBackface": "building-i-back.jpg",
    "name": "Reserva de Piedra",
    "type": "building",
    "x": 0,
    "y": 0,
  },
  Object {
    "faceDown": false,
    "id": "7d710b54-165a-4518-8a2c-ae3b03ef3b6a",
    "imageFile": "",
    "imageFileBackface": "building-i-back.jpg",
    "name": "Reserva de Madera",
    "type": "building",
    "x": 0,
    "y": 0,
  },
  Object {
    "faceDown": false,
    "id": "20480960-e205-431d-82de-9a1404c7d255",
    "imageFile": "",
    "imageFileBackface": "building-i-back.jpg",
    "name": "Almacém de Piedra",
    "type": "building",
    "x": 0,
    "y": 0,
  },
  Object {
    "faceDown": false,
    "id": "2437d5a9-17af-4f64-99f3-b6e390dbb295",
    "imageFile": "",
    "imageFileBackface": "building-i-back.jpg",
    "name": "Taberna",
    "type": "building",
    "x": 0,
    "y": 0,
  },
  Object {
    "faceDown": false,
    "id": "be450573-2a52-44e6-a87a-895cb605ba2f",
    "imageFile": "",
    "imageFileBackface": "building-i-back.jpg",
    "name": "Altar",
    "type": "building",
    "x": 0,
    "y": 0,
  },
]
`;

exports[`getShuffledCards shuffle second age cards 1`] = `
Array [
  Object {
    "faceDown": false,
    "id": "76b151d4-7607-47fd-a106-8999c8337306",
    "imageFile": "",
    "imageFileBackface": "building-ii-back.jpg",
    "name": "Guarnicion",
    "type": "building",
    "x": 0,
    "y": 0,
  },
  Object {
    "faceDown": false,
    "id": "923b24ba-e40b-4e4b-913c-42880ca8422d",
    "imageFile": "",
    "imageFileBackface": "building-ii-back.jpg",
    "name": "Reserva de Arcilla",
    "type": "building",
    "x": 0,
    "y": 0,
  },
  Object {
    "faceDown": false,
    "id": "8d3c597f-bc55-46a6-b9b3-011d3e0908eb",
    "imageFile": "",
    "imageFileBackface": "building-ii-back.jpg",
    "name": "Taberna",
    "type": "building",
    "x": 0,
    "y": 0,
  },
  Object {
    "faceDown": false,
    "id": "b0fbff14-848d-4c03-9444-d81fe720ffd9",
    "imageFile": "",
    "imageFileBackface": "building-ii-back.jpg",
    "name": "Scriptorium",
    "type": "building",
    "x": 0,
    "y": 0,
  },
  Object {
    "faceDown": false,
    "id": "5e43239c-86e2-464d-8c77-1b3e658b742c",
    "imageFile": "",
    "imageFileBackface": "building-ii-back.jpg",
    "name": "Reserva de Piedra",
    "type": "building",
    "x": 0,
    "y": 0,
  },
  Object {
    "faceDown": false,
    "id": "d282b2c1-acd1-4990-89db-14e174ec3e24",
    "imageFile": "",
    "imageFileBackface": "building-ii-back.jpg",
    "name": "Altar",
    "type": "building",
    "x": 0,
    "y": 0,
  },
  Object {
    "faceDown": false,
    "id": "2dcb2530-6ec6-479a-a104-60815a90676b",
    "imageFile": "",
    "imageFileBackface": "building-ii-back.jpg",
    "name": "Torre de Guardia",
    "type": "building",
    "x": 0,
    "y": 0,
  },
  Object {
    "faceDown": false,
    "id": "925201b2-844d-4702-9a88-962690f0398f",
    "imageFile": "",
    "imageFileBackface": "building-ii-back.jpg",
    "name": "Campo Maderero",
    "type": "building",
    "x": 0,
    "y": 0,
  },
  Object {
    "faceDown": false,
    "id": "a4be6db9-132c-4b02-a430-1b6edcc24f2c",
    "imageFile": "",
    "imageFileBackface": "building-ii-back.jpg",
    "name": "Almacém de Arcilla",
    "type": "building",
    "x": 0,
    "y": 0,
  },
  Object {
    "faceDown": false,
    "id": "1c6e095b-b203-4dd0-81ef-c315e6384353",
    "imageFile": "",
    "imageFileBackface": "building-ii-back.jpg",
    "name": "Farmacia",
    "type": "building",
    "x": 0,
    "y": 0,
  },
  Object {
    "faceDown": false,
    "id": "cc888a4f-769d-4999-8587-62c24513a2b0",
    "imageFile": "",
    "imageFileBackface": "building-ii-back.jpg",
    "name": "Almacém de Madera",
    "type": "building",
    "x": 0,
    "y": 0,
  },
  Object {
    "faceDown": false,
    "id": "cdd2726b-3855-4926-8bb1-5d9d21279122",
    "imageFile": "",
    "imageFileBackface": "building-ii-back.jpg",
    "name": "Almacém de Piedra",
    "type": "building",
    "x": 0,
    "y": 0,
  },
  Object {
    "faceDown": false,
    "id": "df4aee8d-f613-4f2c-8143-f4bcc40b1d42",
    "imageFile": "",
    "imageFileBackface": "building-ii-back.jpg",
    "name": "Cantera",
    "type": "building",
    "x": 0,
    "y": 0,
  },
  Object {
    "faceDown": false,
    "id": "8e1d882a-89a2-4bdc-ae1e-881450255b4b",
    "imageFile": "",
    "imageFileBackface": "building-ii-back.jpg",
    "name": "Palenque",
    "type": "building",
    "x": 0,
    "y": 0,
  },
  Object {
    "faceDown": false,
    "id": "bdbc6907-b107-4db7-8944-346ad0636b8d",
    "imageFile": "",
    "imageFileBackface": "building-ii-back.jpg",
    "name": "Explotación de Arcilla",
    "type": "building",
    "x": 0,
    "y": 0,
  },
  Object {
    "faceDown": false,
    "id": "34d6f668-8a6b-43fc-b8c5-b160b68b959b",
    "imageFile": "",
    "imageFileBackface": "building-ii-back.jpg",
    "name": "Reserva de Madera",
    "type": "building",
    "x": 0,
    "y": 0,
  },
  Object {
    "faceDown": false,
    "id": "c7304766-82b7-4ad2-bfd9-537280be9902",
    "imageFile": "",
    "imageFileBackface": "building-ii-back.jpg",
    "name": "Prensa",
    "type": "building",
    "x": 0,
    "y": 0,
  },
  Object {
    "faceDown": false,
    "id": "ef985d21-ba8e-4ba2-8e14-97b9d1a889c3",
    "imageFile": "",
    "imageFileBackface": "building-ii-back.jpg",
    "name": "Cristalería",
    "type": "building",
    "x": 0,
    "y": 0,
  },
  Object {
    "faceDown": false,
    "id": "16b1e0d7-b73e-4575-a61c-3d65b916f1d3",
    "imageFile": "",
    "imageFileBackface": "building-ii-back.jpg",
    "name": "Teatro",
    "type": "building",
    "x": 0,
    "y": 0,
  },
  Object {
    "faceDown": false,
    "id": "540539ae-e089-4f2f-815e-b3947797610d",
    "imageFile": "",
    "imageFileBackface": "building-ii-back.jpg",
    "name": "Taller",
    "type": "building",
    "x": 0,
    "y": 0,
  },
]
`;

exports[`getShuffledCards shuffle third age/guild cards 1`] = `
Array [
  Object {
    "faceDown": false,
    "id": "17d6469f-4223-428f-bb21-9a8bbb06a533",
    "imageFile": "",
    "imageFileBackface": "building-iii-back.jpg",
    "name": "Reserva de Madera",
    "type": "building",
    "x": 0,
    "y": 0,
  },
  Object {
    "faceDown": false,
    "id": "3fd7ac0e-7e1c-405a-9faf-f128535c942c",
    "imageFile": "",
    "imageFileBackface": "building-g-back.jpg",
    "name": "Gremio de Magistrados",
    "type": "building",
    "x": 0,
    "y": 0,
  },
  Object {
    "faceDown": false,
    "id": "fcfd197a-a55f-4252-8753-8b9b50d1ddc8",
    "imageFile": "",
    "imageFileBackface": "building-iii-back.jpg",
    "name": "Teatro",
    "type": "building",
    "x": 0,
    "y": 0,
  },
  Object {
    "faceDown": false,
    "id": "1023003d-26a4-4b2f-884e-92b4de034dd0",
    "imageFile": "",
    "imageFileBackface": "building-iii-back.jpg",
    "name": "Altar",
    "type": "building",
    "x": 0,
    "y": 0,
  },
  Object {
    "faceDown": false,
    "id": "2f95b0a3-d1bf-4f6e-a76f-1d29f34067cc",
    "imageFile": "",
    "imageFileBackface": "building-iii-back.jpg",
    "name": "Cantera",
    "type": "building",
    "x": 0,
    "y": 0,
  },
  Object {
    "faceDown": false,
    "id": "e7c32983-07ba-43c4-bd7c-dc89731ab90a",
    "imageFile": "",
    "imageFileBackface": "building-iii-back.jpg",
    "name": "Almacém de Madera",
    "type": "building",
    "x": 0,
    "y": 0,
  },
  Object {
    "faceDown": false,
    "id": "2a71a8c2-6969-4cb6-bd42-a516b00d12e2",
    "imageFile": "",
    "imageFileBackface": "building-iii-back.jpg",
    "name": "Reserva de Arcilla",
    "type": "building",
    "x": 0,
    "y": 0,
  },
  Object {
    "faceDown": false,
    "id": "cf4b069a-e753-4401-a51c-36079e2d57c8",
    "imageFile": "",
    "imageFileBackface": "building-g-back.jpg",
    "name": "Gremio de Construciones",
    "type": "building",
    "x": 0,
    "y": 0,
  },
  Object {
    "faceDown": false,
    "id": "0edfebf5-545c-41f2-8a3d-5edd48c49189",
    "imageFile": "",
    "imageFileBackface": "building-iii-back.jpg",
    "name": "Establos",
    "type": "building",
    "x": 0,
    "y": 0,
  },
  Object {
    "faceDown": false,
    "id": "25e819e3-fab6-40c2-9534-5d9582d6152d",
    "imageFile": "",
    "imageFileBackface": "building-g-back.jpg",
    "name": "Gremio de Científicos",
    "type": "building",
    "x": 0,
    "y": 0,
  },
  Object {
    "faceDown": false,
    "id": "21bf536b-c02c-444c-9e06-667e00866a81",
    "imageFile": "",
    "imageFileBackface": "building-iii-back.jpg",
    "name": "Torre de Guardia",
    "type": "building",
    "x": 0,
    "y": 0,
  },
  Object {
    "faceDown": false,
    "id": "74301971-6be8-4fc4-90c8-da47b10daf67",
    "imageFile": "",
    "imageFileBackface": "building-iii-back.jpg",
    "name": "Baños",
    "type": "building",
    "x": 0,
    "y": 0,
  },
  Object {
    "faceDown": false,
    "id": "de0e8e2d-d65a-4c79-a701-9a476e4a7121",
    "imageFile": "",
    "imageFileBackface": "building-iii-back.jpg",
    "name": "Taberna",
    "type": "building",
    "x": 0,
    "y": 0,
  },
  Object {
    "faceDown": false,
    "id": "63e7dd2a-c473-4d3e-8aa5-d852b812d80a",
    "imageFile": "",
    "imageFileBackface": "building-iii-back.jpg",
    "name": "Guarnicion",
    "type": "building",
    "x": 0,
    "y": 0,
  },
  Object {
    "faceDown": false,
    "id": "29713bb6-d892-4e13-a1d8-72076693dc53",
    "imageFile": "",
    "imageFileBackface": "building-iii-back.jpg",
    "name": "Almacém de Piedra",
    "type": "building",
    "x": 0,
    "y": 0,
  },
  Object {
    "faceDown": false,
    "id": "e9d049d4-96f8-4538-afe7-132f34bafa36",
    "imageFile": "",
    "imageFileBackface": "building-iii-back.jpg",
    "name": "Explotación de Arcilla",
    "type": "building",
    "x": 0,
    "y": 0,
  },
  Object {
    "faceDown": false,
    "id": "4896af85-c851-4302-b679-b3f62cda7413",
    "imageFile": "",
    "imageFileBackface": "building-iii-back.jpg",
    "name": "Almacém de Arcilla",
    "type": "building",
    "x": 0,
    "y": 0,
  },
  Object {
    "faceDown": false,
    "id": "0b5d4f9f-577b-4032-bcae-b17900877b0e",
    "imageFile": "",
    "imageFileBackface": "building-iii-back.jpg",
    "name": "Campo Maderero",
    "type": "building",
    "x": 0,
    "y": 0,
  },
  Object {
    "faceDown": false,
    "id": "f2a03138-7b25-493e-bca0-f67450dbbf22",
    "imageFile": "",
    "imageFileBackface": "building-iii-back.jpg",
    "name": "Cristalería",
    "type": "building",
    "x": 0,
    "y": 0,
  },
  Object {
    "faceDown": false,
    "id": "cb34d819-f670-4102-a131-33e635934167",
    "imageFile": "",
    "imageFileBackface": "building-iii-back.jpg",
    "name": "Palenque",
    "type": "building",
    "x": 0,
    "y": 0,
  },
]
`;

exports[`getShuffledCards shuffle third age/guild cards 2`] = `
Array [
  Object {
    "faceDown": false,
    "id": "87d2c9d5-374f-417e-8916-c1ad0db104af",
    "imageFile": "",
    "imageFileBackface": "building-g-back.jpg",
    "name": "Gremio de Científicos",
    "type": "building",
    "x": 0,
    "y": 0,
  },
  Object {
    "faceDown": false,
    "id": "8ebeb4c1-d118-43fa-a97b-95f79a4e1419",
    "imageFile": "",
    "imageFileBackface": "building-g-back.jpg",
    "name": "Gremio de Armadores",
    "type": "building",
    "x": 0,
    "y": 0,
  },
  Object {
    "faceDown": false,
    "id": "5360ddd6-aee9-4f23-9779-39007664a813",
    "imageFile": "",
    "imageFileBackface": "building-g-back.jpg",
    "name": "Gremio de Construciones",
    "type": "building",
    "x": 0,
    "y": 0,
  },
  Object {
    "faceDown": false,
    "id": "590d5f83-f2ac-4663-9b37-622ee85be16c",
    "imageFile": "",
    "imageFileBackface": "building-g-back.jpg",
    "name": "Gremio de Magistrados",
    "type": "building",
    "x": 0,
    "y": 0,
  },
  Object {
    "faceDown": false,
    "id": "bce3a77e-6603-40bc-8bc7-994f73a84487",
    "imageFile": "",
    "imageFileBackface": "building-g-back.jpg",
    "name": "Gremio de Mercaderes",
    "type": "building",
    "x": 0,
    "y": 0,
  },
  Object {
    "faceDown": false,
    "id": "6c48145d-5b65-4c2e-b66f-069d0008e047",
    "imageFile": "",
    "imageFileBackface": "building-g-back.jpg",
    "name": "Gremio de Prestamistas",
    "type": "building",
    "x": 0,
    "y": 0,
  },
  Object {
    "faceDown": false,
    "id": "34e5b221-38a7-45ac-b2c2-56d5b4c3b4b2",
    "imageFile": "",
    "imageFileBackface": "building-g-back.jpg",
    "name": "Gremio de Tácticos",
    "type": "building",
    "x": 0,
    "y": 0,
  },
  Object {
    "faceDown": false,
    "id": "00a6c516-b197-48f7-bdb9-00e5835fe307",
    "imageFile": "",
    "imageFileBackface": "building-g-back.jpg",
    "name": "Gremio de Magistrados",
    "type": "building",
    "x": 0,
    "y": 0,
  },
  Object {
    "faceDown": false,
    "id": "28fcd751-bdab-478b-ad82-d701f3ea1730",
    "imageFile": "",
    "imageFileBackface": "building-g-back.jpg",
    "name": "Gremio de Construciones",
    "type": "building",
    "x": 0,
    "y": 0,
  },
  Object {
    "faceDown": false,
    "id": "82372ef8-b87b-4d2e-adcd-f0a864633186",
    "imageFile": "",
    "imageFileBackface": "building-g-back.jpg",
    "name": "Gremio de Armadores",
    "type": "building",
    "x": 0,
    "y": 0,
  },
]
`;
